x-pack/plugins/observability/public/pages/slo_details/hooks/use_fetch_slo_details.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HttpSetup } from '@kbn/core-http-browser';
+import { useCallback, useMemo } from 'react';
+import { useDataFetcher } from '../../../hooks/use_data_fetcher';
+import { SLO } from '../../../typings';
+
+export function useFetchSloDetails(sloId: string): [boolean, SLO | undefined] {
```

Comment:
Most other fetching hooks return an object of the shape `{ <thingThatWasFetched>, loading, error }`. Might be good for consistency to keep that shape?